airflow/api_connexion/security.py

```diff
@@ -32,8 +32,8 @@
     PoolDetails,
     VariableDetails,
 )
+from airflow.providers.fab.www.extensions.init_auth_manager import get_auth_manager
```

Comment:
I think we need to have two `get_auth_manager` because we need to init the auth manager twice:
- One (the main one) in `airflow/api_fastapi/app.py` that is used across the application. This auth manager is init with no parameter as it should be
- One in `providers/src/airflow/providers/fab/www/extensions/init_auth_manager.py`. This auth manager is init with the appbuilder as parameter. The appbuilder is used across the FAB provider codebase for many different features. 

Overall, `get_auth_manager` from `airflow/api_fastapi/app.py` should be used everywhere in the codebase but FAB provider. `get_auth_manager` from `providers/src/airflow/providers/fab/www/extensions/init_auth_manager.py` should be used only in FAB provider.

I am aware this is confusing. Maybe renaming `get_auth_manager` to something else like `get_fab_auth_manager` in the fab provider would make it less confusing?

Reply:
Hmm, why do we need 2 get_auth_managers? Seems like the fab auth manager also needs to be used in fastapi, if that's the auth manager that was configured. Are there differences in the way its used by fastapi and fab that necessitates a fastapi-fab-auth-manager and a fab-fab-auth-manager?

I realize the fab auth manager might have to get a little creative, since it doesn't really make sense for the auth manager interface to take an appbuilder any longer... but I imagine that is something we can work through.

Reply:
You are right. We should have only one, I updated the PR to only have one auth manager across the different applications (fastapi application, legacy Flask application and Flask application in the FAB provider). For both Flask applications we need to pass the appbuilder to the auth manager but instead of passing it as parameter to the constructor, I just set it like `auth_manager.appbuilder = ...`. It is simpler that way. I also removed `appbuilder` from the interface constructor.

Thanks for calling it out, it is definitely simpler and better that way!

Reply:
Nice 👍. Thats simpler than I was envisioning, love when it goes that way.